src/dynimage.rs

```diff
@@ -658,14 +658,22 @@ impl DynamicImage {
     }
 
     /// Resize this image using the specified filter algorithm.
-    /// Returns a new image. The image's aspect ratio is preserved.
     /// The image is scaled to the maximum possible size that fits
     /// within the bounds specified by `nwidth` and `nheight`.
+    /// The image's aspect ratio is preserved.
+    /// Returns a new image if the requested image size is different than the already existing one.
+    /// Otherwise the function returns a copy of self.
```

Comment:
Other than performance, the caller shouldn't notice a difference, right? If so, I think we should leave off the discussion of returning a new image vs. a copy. That way users don't have to think about it